workflow/util/util.go

```diff
@@ -971,7 +971,7 @@ func FormulateRetryWorkflow(ctx context.Context, wf *wfv1.Workflow, restartSucce
 				log.Debugf("Reset %s node %s since it's a group node", node.Name, string(node.Phase))
 				continue
 			} else {
-				if isDescendantNodeSucceeded(wf, node, nodeIDsToReset) {
+				if node.Type == wfv1.NodeTypePod && isDescendantNodeSucceeded(wf, node, nodeIDsToReset) {
```

Comment:
It feels like this isn't the whole list of NodeTypes that should be considered here.
HTTP, Plugin and Container come to mind, but I haven't thought it through completely.

Reply:
Oh, I originally wrote it as not equal to retry, but there was a test error, and now I changed it back.